### PR TITLE
Reqtorio Expansion V 1.2

### PR DESCRIPTION
**Pull request by Naaanii**
## About The Pull Request
Further Expands Reqtorio and reworks the howitzer recipe to use the fancy new machines.

Adds more material sprites for more combinations as well. Sprites thanks to [aGhost!](https://github.com/aGhostCoder)!

Recipes added:
SWAT masks (assembly cost 50 points for 20 pieces)
<s>Optical imagers (assembly cost 50 points for 20 pieces)</s>
Trauma and Burn kits and Splints (assembly cost 50 for 15 pieces)
All req modules [Valk, Mimir 2, Tyr, Hlin, Surt] (assembly cost 60 for 10 pieces)
## Why It's Good For The Game
More recipes
More combos
More profit
More fun.
## Changelog
:cl:
add: Added more recipes for items that can be made via reqtorio. (SWAT masks, <s>Optical Imagers,</s> Burn and Trauma Kits and Splints, Valk, Mimir, Tyr, Hlin and Surt modules)
add: Added more material sprites. (See factoryparts.dmi)
/:cl:

### CODE REVIEW

**Review by Kuro020 — APPROVED** [review]

File: code/modules/reqs/supplypacks.dm [hunk 1]
```diff
@@ -1806,3 +1806,48 @@ FACTORY
...
+	contains = list(/obj/item/factory_refill/module_tyr2_refill)
+	cost = 60
+
+/datum/supply_packs/factory/module_hlin_refill
+	name = "Hlin Explosive Compensation Module assembly refill"
+	contains = list(/obj/item/factory_refill/module_hlin_refill)
+	cost = 60
+
+/datum/supply_packs/factory/module_surt_refill
+	name = "Hades Incendiary Insulation System assembly refill"
```
Comment (Arctrooper8) [inline code/modules/reqs/supplypacks.dm:1851]: Called Surt refill in item path and used to make Surt modules yet it's named Hades?

File: code/modules/factory/parts.dm [hunk 2]
```diff
@@ -316,3 +316,113 @@ GLOBAL_LIST_INIT(howitzer_shell, list(
...
+	result = /obj/item/storage/belt/rig/medical
+
+/obj/item/factory_part/med_rig/Initialize()
+	. = ..()
+	recipe = GLOB.meds
+
+/obj/item/factory_part/med_lifesaver
+	name = "M276 pattern lifesaver bag"
+	desc = "An unfinished M276 pattern lifesaver bag"
+	result = /obj/item/storage/belt/lifesaver/full
```
Comment (Arctrooper8) [inline code/modules/factory/parts.dm:370]: I don't know if allowing full medical belts to be made is a good thing balance wise, but I don't exactly see it being used much so it shouldn't be a major issue.

**Comment by KerbalCitizen** [comment]
Are the medical belt and lifesaver belts filled when they get produced or just empty?

File: code/modules/factory/parts.dm (hunk 2 above)
Comment (Naaanii) [reply code/modules/factory/parts.dm]: Mass production of meds seems alright. Most medbelts aren't even used shipside, this just offers the option and is quite expensive though through mass production the cost makes up for itself.

**Comment by MereWalker** [comment]
Medical belts on demand! Medical ground side is saved!

**Comment by zzurge** [comment]
zased

**Comment by Serxule** [comment]
i dunno if jaeger modules is a good idea (might end up being a bit too strong) but swat masks + imagers is a great idea since those are constantly in high demand

File: code/modules/factory/parts.dm (hunk 2 above)
Comment (Alexeivichi) [reply code/modules/factory/parts.dm]: > Mass production of meds seems alright. Most medbelts aren't even used shipside, this just offers the option and is quite expensive though through mass production the cost makes up for itself.


This is actually one hell of a balance shake up. First thing was medication being limited yet requiring a lot of annoying steps to put em' planetside AND make it easy to carry. But providing a full belt with ease through factorio would allow marines to sustain themselves far easier.

I won't say too much against it but I still don't think it's that much of a good idea unless we're going for a faster pace of combat which... TGMC has been going down the road of.

Comment (Naaanii) [reply code/modules/factory/parts.dm]: Changed belts to kits+splints

**Comment by Arctrooper8** [comment]
Splints are free and infinite so it feels weird having the factory make any at all.

File: code/modules/factory/parts.dm [hunk 3]
```diff
@@ -27,8 +27,9 @@
 /obj/item/factory_part/proc/advance_stage()
 	stage++
 	if(length(recipe) < stage)
-		GLOB.round_statistics.req_items_produced[result]++
-		new result(loc)
+		for(var/production AS in result)
+			GLOB.round_statistics.req_items_produced[production]++
+			new production(loc)
```
Comment (TiviPlus) [inline code/modules/factory/parts.dm:32]: this should support result as both list and typepath instead as initializing uneeded lists would be cpu waste

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/factory/parts.dm (hunk 3 above)
Comment (Naaanii) [reply code/modules/factory/parts.dm]: Done

**Comment by Fikou** [comment]
what the hell

File: code/modules/factory/parts.dm [hunk 4]
```diff
@@ -97,7 +102,7 @@ GLOBAL_LIST_INIT(sadar_ammo_recipe, list(
 /obj/item/factory_part/sadar_ap
 	name = "SADAR AP missile asssembly"
 	desc = "An unfinished sleek missile with an AP warhead."
-	result = /obj/item/ammo_magazine/rocket/sadar/ap
+	result = list(/obj/item/ammo_magazine/rocket/sadar/ap)
```
Comment (TiviPlus) [inline code/modules/factory/parts.dm:105]: Undo the changes making the single result to lists

Comment (Naaanii) [reply code/modules/factory/parts.dm]: Done
